api/src/main/java/com/datastrato/gravitino/rel/SupportsPartitions.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel;
+
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
+import com.datastrato.gravitino.rel.expressions.partitions.Partition;
+import java.util.Map;
+
+public interface SupportsPartitions extends Table {
+
+  /**
+   * List all partition names of the table
+   *
+   * @return The list of partition names
+   */
+  String[] listPartitionNames();
+
+  /**
+   * List all partitions
+   *
+   * @return The list of partitions
+   */
+  Partition[] listPartitions();
+
+  /**
+   * get a partition by partition name
+   *
+   * @param partitionName the name of the partition
+   * @return the partition
+   * @throws NoSuchPartitionException if the partition does not exist
+   */
+  Partition getPartition(String partitionName) throws NoSuchPartitionException;
+
+  /**
+   * Check if a partition exists.
+   *
+   * @param partitionName The name of the partition.
```

Comment:
please description what's a partitionName? like 'year=2022'? this's hive format, it is general?
 Should we use a specify PartiionName to represent partitionName?

Reply:
we should not limit the partition name format. 

Partition names can be generated in two ways: one is specified by the user when adding a partition, and the other is automatically generated by the engine.